media-libs/ffmpegsource/ffmpegsource-2.23.ebuild

```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit autotools eutils flag-o-matic vcs-snapshot
+
+DESCRIPTION="A libav/ffmpeg based source library for easy frame accurate access"
+HOMEPAGE="https://github.com/FFMS/ffms2"
+SRC_URI="https://github.com/FFMS/ffms2/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0/4"
+KEYWORDS="~amd64 ~x86"
+IUSE="libav"
+
+RDEPEND="
+	sys-libs/zlib
+	!libav? ( >=media-video/ffmpeg-2.4:0= )
+	libav? ( >=media-video/libav-9:0= )
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+"
+
+pkg_pretend() {
+	if [[ ${MERGE_TYPE} != "binary" ]] && ! test-flag-CXX -std=c++11; then
```

Comment:
In general, `pkg_pretend` is just a "early warning failure" thing. If you do use `pkg_pretend`, also always do the same in `pkg_setup`, as you have to be sure that compiler hasn't lost the capabilities you were testing for, preferably by sticking everything into a function and calling that function twice (never call phase functions explicitly!)